modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -921,37 +912,17 @@ private int txCatalogVersion(HybridTimestamp beginTs) {
         return catalogService.activeCatalogVersion(beginTs.longValue());
     }
 
-    /**
-     * Returns the transaction catalog version.
-     *
-     * @param txId Transaction ID.
-     */
-    private int txCatalogVersion(UUID txId) {
-        return txCatalogVersion(TransactionIds.beginTimestamp(txId));
+    /** See {@link RwTxFinishedHandler#isRwTransactionsFinished(int)}. */
+    boolean isRwTransactionsFinished(int catalogVersion) {
```

Comment:
Do we need it here just for tests? Can't the test be rewritten to actually test `RwTxFinishedHandler`? (because now this method just delegates to the corresponding method in `RwTxFinishedHandler`)

Reply:
The test that uses is an integration test, so what I suggested is probably not too relevant.

Reply:
Yes, this method is now used only in integration tests; unit tests do not conveniently cover the logic described in the ticket. I don’t know the best way to test the logic from the ticket; making the handler external doesn’t seem like the right approach. If you have a suggestion, then welcome.

Reply:
Fix it

Reply:
Fix it